src/Bridge/Doctrine/EventListener/PublishMercureUpdatesListener.php

```diff
@@ -225,7 +226,7 @@ private function publishUpdate($object, array $options, string $type): void
         $updates = array_merge([$this->buildUpdate($iri, $data, $options)], $this->getGraphQlSubscriptionUpdates($object, $options, $type));
 
         foreach ($updates as $update) {
-            $this->messageBus ? $this->dispatch($update) : ($this->publisher)($update);
+            $this->mercure->getPublisher($options['hub'] ?? null)->publish($update);
```

Comment:
@dunglas do we wanna keep support for message bus here?

Reply:
good question, as the class is experimental I think that we can do pretty much what we want :)

Reply:
the thing with messenger is that we are not gaining anything from using it, we are replacing an http request, but a request to rabbitmq, redis, mysql server .. etc. which would later result in an http request.

Reply:
I propose to deprecate it (but we must provide an upgrade path).

Reply:
how do you propose we deprecate this?

Reply:
Why not triggering a deprecation in an if?

Reply:
that means you will get the deprecation notice if you have messenger installed, so there's no way for you to get around the deprecation.

Reply:
Right. Then we would need to introduce a configuration parameter, like `enable_async_update` or something like this, make it `true` by default, trigger a deprecation if not set to `false`.
But maybe it's too much work for an experimental component :sweat_smile:
I let @dunglas decide but personally I'm OK for a BC break mentioned in the CHANGELOG.